src/features/serde/mod.rs

```diff
@@ -76,6 +138,20 @@ impl serde_incl::de::Error for crate::error::DecodeError {
     }
 }
 
+#[cfg(not(feature = "std"))]
+impl serde_incl::de::StdError for crate::error::DecodeError {}
+
+#[cfg(not(feature = "alloc"))]
+impl serde_incl::de::Error for crate::error::DecodeError {
+    fn custom<T>(_: T) -> Self
+    where
+        T: core::fmt::Display,
+    {
+        DecodeError::CustomError.into()
+    }
+}
+
+#[cfg(feature = "alloc")]
```

Comment:
Why is this _with_ alloc when the above is _without_?

Reply:
I noticed the `DecodeError` impls were below the `EncodeError` impl, fixed